cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Broker.java

```diff
@@ -583,22 +588,70 @@ void setReplicaLoad(TopicPartition tp, AggregatedMetricValues aggregatedMetricVa
     return brokerMap;
   }
 
+  /**
+   * Get per-logdir disk utilization information of the broker.
+   *
+   * @return The per-logdir disk utilization information. If information of replica placement over disk is not populated,
+   *         return null.
+   */
+  public Map<String, Double> diskUtils() {
+    if (_diskByLogdir.isEmpty()) {
+      return null;
+    }
+    if (_diskByLogdir.values().stream().mapToInt(disk -> disk.replicas().size()).sum() == 0) {
+      return null;
+    }
+    Map<String, Double> diskUtilMap = new HashMap<>(_diskByLogdir.size());
+    for (Map.Entry<String, Disk> entry : _diskByLogdir.entrySet()) {
+      diskUtilMap.put(entry.getKey(), entry.getValue().utilization());
+    }
+    return diskUtilMap;
+  }
+
+  /**
+   * Get per-logdir disk capacity information of the broker.
+   *
+   * @return The per-logdir disk capacity information. If information of replica placement over disk is not populated,
+   *         return null.
+   */
+  public Map<String, Double> diskCapacities() {
+    if (_diskByLogdir.isEmpty()) {
+      return null;
+    }
+    if (_diskByLogdir.values().stream().mapToInt(disk -> disk.replicas().size()).sum() == 0) {
```

Comment:
Do we need this check -- is it possible to double check our assumptions about inputs and also update the relevant constructor JavaDoc of `BrokerCapacityInfo`?

Reply:
change the logic to use newly added `_replicaPlacementInfoPopulated` flag.

Reply:
Isn't it enough to check `if (_diskByLogdir.isEmpty()) {`? If so, we can remove `_replicaPlacementInfoPopulated` completely.